### ISSUE TITLE
ZIP archive of Dart-SDK API missing details in the right navigation column (v3.1.0 to v.3.2.6)

### ISSUE THREAD

**Issue by hoylen** [dart-lang/sdk#54721]
The ZIP archive download of the Dart-SDK API (downloaded from <https://dart.dev/get-dart/archive>) no longer shows navigation links for a library or class.

The documentation [ZIP for v3.0.7](https://storage.googleapis.com/dart-archive/channels/stable/release/3.0.7/api-docs/dartdocs-gen-api.zip) and earlier has three columns: a left navigation column listing the libraries, the main content, and a right navigation column listing the library's classes, constants, functions, etc.

The details in the third column is now **missing** from the recent ZIP files. The right column in [ZIP for v3.1.0](https://storage.googleapis.com/dart-archive/channels/stable/release/3.1.0/api-docs/dartdocs-gen-api.zip) and newer (v3.2.6 at the time of writing) does not have any links in the right navigation column. It just shows the name of the library.

* Dart version and tooling diagnostic info
    * Dart 3.2.5 (but it is irrelevant to this issue)
* Whether you are using Windows, macOS, or Linux (if applicable)
    * macOS (archive extracted using command line "unzip" program, as well as double clicking on the ZIP file's icon in the Finder)
* Whether you are using Chrome, Safari, Firefox, Edge (if applicable)
    * Problem occurs with Safari, Firefox and Chrome.

**Comment by parlough** [comment]
Thanks for reporting this @hoylen! These versions switched to a shared sidebar to reduce the size of generated documentation for large packages and SDKs. As a result, to view them with the sidebars, you'd need to serve the contents with an HTTP server.

A quick way to do so would be to use [`package:dhttpd`](https://pub.dev/packages/dhttpd) from the unzipped directory:

```terminal
dart pub global activate dhttpd
dart pub global run dhttpd
```

Then open the link it prints out in your browser of choice.

As a note, you can also view the API docs for specific versions on api.dart.dev. For example you can find the API docs for Dart 3.1.1 at https://api.dart.dev/stable/3.1.1/index.html.

If you're able to share, I am curious what you're using the downloaded version of the SDK docs for. Perhaps your workflow can be better supported :)

\cc @srawlins This is due to the sidenav change. Do you think the solution for now is to just document that it's required to serve the docs?

**Comment by srawlins** [comment]
> Do you think the solution for now is for us to document on dart.dev that it's required to serve the docs?

Yes! Sorry I probably didn't do a good look-around to see what text should be updated.

**Comment by hoylen** [comment]
So that's why it behaves differently. Thanks for the quick reply.

Definitely documenting the need for a Web server is necessary. But most people won't find the documentation when they come across the problem, or know where to look for it. Two possible solutions to that are:

1. Have the right navigation column show some static HTML text that says something like, "Please serve this documentation from a Web server to use this navigation column" and have it automatically replaced by the real links if it is served up by a Web server. Could some simple JavaScript do that?
2. The API documentation ZIP archive contain a README file to explain how to install it.

My use case is pretty simple: to access the documentation without needing network access. Because a local copy is slightly faster and to avoid wasting bandwidth. Running a local Web server is not a huge problem, though it is less convenient than simply opening up files in a Web browser.

It appears part of the problem is the browser's _Cross-Origin Resource Sharing_ (CORS) mechanism blocking the page from loading the index.json and sidebar HTML files. I'm not familiar with how CORS works with "file:" URIs or what has changed from the JavaScript generated for the older documentation. Has anyone looked into that (as an alternative to running a Web server)?

**Comment by hoylen** [comment]
I now see why CORS is causing the problem. The new _docs.dart.js_ file invokes `XMLHttpRequest` and `fetch`, both of which involve CORS. Browser implementations differ on how CORS behaves with "file" scheme URIs, but they err on the side of security and usually block the requests (even though you might assume a file loading another file should be ok). There is a [workaround](https://stackoverflow.com/questions/63185365/does-xmlhttprequest-accept-the-file-protocol) flag for running _Google Chrome_, but it is inconvenient and risky to use.

The older version of _docs.dart.js_ does not use _XMLHttpRequest_. However, it did use _fetch_. But I haven't looked into why the old invocation of _fetch_ appears to work and the new one does not.

Sounds like serving up the files with a local Web browser is the best approach... as long as the user downloading the ZIP file somehow knows that is how it must be used.
